qf_lib/backtesting/trading_session/backtest_trading_session.py

```diff
@@ -82,6 +82,15 @@ def use_data_preloading(self, tickers: Union[Ticker, Sequence[Ticker]], time_del
         # The tickers and price fields are sorted in order to always return the same hash of the data bundle for
         # the same set of tickers and fields
         tickers, _ = convert_to_list(tickers, Ticker)
+
+        if self.portfolio.currency is not None:
+            currencies = set([ticker.currency for ticker in tickers])
+            if any(currency != self.portfolio.currency for currency in currencies):
```

Comment:
To simplify the flow a little bit we could:
1. First we discard portfolio currency from currencies
2. We check if the set is empty or not